tests/test_machine.py

```diff
@@ -1,69 +1,82 @@
+
+    def _callTestMethod(self, method: str) -> None:
+        try:
+            super()._callTestMethod(method)
+        except Exception as e:
+            self.exc_info = exc_info()
+            raise e
+
+    def __init_subclass__(cls, args: List[str], reboot: bool = True) -> None:
+        cls.reboot = reboot
+        cls.args = args
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.machine = Machine(cls.args)
+        cls.machine.__enter__()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.machine.__exit__(exc_info)
+
+    def tearDown(self) -> None:
+        if self.reboot:
+            self.machine.reboot_image(exc_info)
+
+class MkosiMachineTester(MkosiMachineTest, args=["boot"], reboot=True):
+    def test_simple_run(self) -> None:
+        process = self.machine.run(["echo", "This is a test."])
+        assert process.stdout.strip("\n") == "This is a test."
+
+    def test_wrong_command(self) -> None:
+        # Check = True from mkosi.backend.run(), therefore we see if an exception is raised
         with pytest.raises(MkosiException):
-            m.run(["NonExisting", "Command"])
+            self.machine.run(["NonExisting", "Command"])
         with pytest.raises(MkosiException):
-            m.run(["ls", "NullDirectory"])
+            self.machine.run(["ls", "NullDirectory"])
 
-    assert m.exit_code == 0
+        # Check = False to see if stderr and returncode have the expected values
+        result = self.machine.run(["NonExisting", "Command"], check=False)
+        assert (result.returncode in (203, 127))
 
-    # Second group of tests with check = False to see if stderr and returncode have the expected values
-    with machine.Machine([verb]) as m:
-        result = m.run(["NonExisting", "Command"], check=False)
-        assert result.returncode in (203, 127)
-
-        result = m.run(["ls", "-"], check=False)
+        result = self.machine.run(["ls", "-"], check=False)
         assert result.returncode == 2
         assert "No such file or directory" in result.stderr
 
-    assert m.exit_code == 0
-
-
-def test_infinite_command(verb: str) -> None:
-    with machine.Machine([verb]) as m:
+    def test_infinite_command(self) -> None:
         with pytest.raises(TimeoutExpired):
-            m.run(["tail", "-f", "/dev/null"], 2)
-
-    assert m.exit_code == 0
+            self.machine.run(["tail", "-f", "/dev/null"], 2)
 
 
-def test_before_boot(verb: str) -> None:
-    m = machine.Machine([verb])
+def test_before_boot() -> None:
+    tmp_machine = Machine(["boot"])
     with pytest.raises(AssertionError):
-        m.run(["ls"])
+        tmp_machine.run(["ls"])
 
 
-def test_after_shutdown(verb: str) -> None:
-    with machine.Machine([verb]) as m:
+def test_after_shutdown() -> None:
+    with Machine(["boot"]) as machine:
         pass
 
     with pytest.raises(AssertionError):
-        m.run(["ls"])
-    assert m.exit_code == 0
```

Comment:
Same here, a bunch of unnecessary changes. Let's keep these the same as much as possible aside from the verb change